### ISSUE TITLE
Export model to XML

### ISSUE THREAD

**Issue by snorfalorpagus** [pywr/pywr#22]
Counterpart to `parse_xml` function.

Need to decide if this should live in the `xmlutils` module, or attached to each component. Attaching to components would make things easier to extend, but means XML code is scattered all over the place.

**Comment by snorfalorpagus** [comment]
Started to implement this in b1dba753578b9d993f95c192acf9f39f202eba63.

* Delegate more objects (e.g. `LicenseCollection`)
* Rename `Parameter` to `ParameterConst` and have both that and `ParameterFunc` inherit from `Parameter`

**Comment by snorfalorpagus** [comment]
@jetuk How should `ParameterFunc` objects be stored?

Suggestion:

```
<parameter type="python" key="max_flow">my_max_flow</parameter>
```

Then in `funcs.py` or similar:

```
def my_max_flow(*args, **kwargs):
    return 42.0
```

Loading would then be something like:
```
import funcs
parameter = ParameterFunc(getattr(funcs, 'my_max_flow'))
```

**Comment by jetuk** [comment]
@snorfalorpagus the above means funcs.py must be saved with the model definition file? Or are you planning to embded it within the xml as well?

What are the alternatives? Save 'my_max_flow' python code in the '<parameter>' tag? Does this make it more or less easy to write an editor?

I guess you could have a another type for simple statements,

    <parameter type="python-inline" key="max_flow">
         return 42.0 if month in (6,7,8) else 50.0
    </parameter>

That might mean you are starting down the road of some sort of domain specific language though.

**Comment by snorfalorpagus** [comment]
I'm not especially keen on storing python code in the XML document. Maybe for very simple functions it would be OK, but for anything more complicated you'd really benefit from the syntax highlighting that editing a pure python file would give you.

**Comment by jetuk** [comment]
Storing is different to editing. I assume you are not wishing users to edit
XML either. No reason an editor couldn't be embedded into the model editor,
but the code stored in XML?

**Comment by snorfalorpagus** [comment]
Closed in favour of #123.
